onnxruntime/contrib_ops/cuda/math/gemm_float8.cu

```diff
@@ -0,0 +1,472 @@
+      return "CUBLAS_STATUS_EXECUTION_FAILED";
+    case CUBLAS_STATUS_INTERNAL_ERROR:
+      return "CUBLAS_STATUS_INTERNAL_ERROR";
+    case CUBLAS_STATUS_NOT_SUPPORTED:
+      return "CUBLAS_STATUS_NOT_SUPPORTED";
+    case CUBLAS_STATUS_LICENSE_ERROR:
+      return "CUBLAS_STATUS_LICENSE_ERROR";
+    default:
+      return "<unknown>";
+  }
+}
+
+static const char* CudaDataTypeToString(cudaDataType_t dt) {
+  switch (dt) {
+    case CUDA_R_16F:
+      return "CUDA_R_16F";
+    case CUDA_R_16BF:
+      return "CUDA_R_16BF";
+    case CUDA_R_32F:
+      return "CUDA_R_32F";
+#if (CUDA_VERSION >= 11080)
+    case CUDA_R_8F_E4M3:
+      return "CUDA_R_8F_E4M3";
+    case CUDA_R_8F_E5M2:
+      return "CUDA_R_8F_E5M2";
+#endif
+    default:
+      return "<unknown>";
+  }
+}
+
+static const char* CublasComputeTypeToString(cublasComputeType_t ct) {
+  switch (ct) {
+    case CUBLAS_COMPUTE_16F:
+      return "CUBLAS_COMPUTE_16F";
+    case CUBLAS_COMPUTE_32F:
+      return "CUBLAS_COMPUTE_32F";
+    case CUBLAS_COMPUTE_32F_FAST_16F:
+      return "CUBLAS_COMPUTE_32F_FAST_16F";
+    case CUBLAS_COMPUTE_32F_FAST_16BF:
+      return "CUBLAS_COMPUTE_32F_FAST_16BF";
+    case CUBLAS_COMPUTE_32F_FAST_TF32:
+      return "CUBLAS_COMPUTE_32F_FAST_TF32";
+    case CUBLAS_COMPUTE_64F:
+      return "CUBLAS_COMPUTE_64F";
+    default:
+      return "<unknown>";
+  }
+}
+
+// It must exist somewhere already.
+cudaDataType_t ToCudaDataType(int32_t element_type) {
+  switch (element_type) {
+    case ONNX_NAMESPACE::TensorProto_DataType_FLOAT:
+      return CUDA_R_32F;
+    case ONNX_NAMESPACE::TensorProto_DataType_FLOAT16:
+      return CUDA_R_16F;
+    case ONNX_NAMESPACE::TensorProto_DataType_BFLOAT16:
+      return CUDA_R_16BF;
+#if (!defined(DISABLE_FLOAT8_TYPES) && (CUDA_VERSION >= 11080))
+    case ONNX_NAMESPACE::TensorProto_DataType_FLOAT8E4M3FN:
+      return CUDA_R_8F_E4M3;
+    case ONNX_NAMESPACE::TensorProto_DataType_FLOAT8E5M2:
+      return CUDA_R_8F_E5M2;
+#endif
+    default:
+      ORT_THROW("Unexpected element_type=", element_type, ".");
+  }
+}
+
+// It must exist somewhere already.
+int32_t TypeSize(int32_t element_type) {
+  switch (element_type) {
+    case ONNX_NAMESPACE::TensorProto_DataType_FLOAT:
+      return 4;
+    case ONNX_NAMESPACE::TensorProto_DataType_BFLOAT16:
+    case ONNX_NAMESPACE::TensorProto_DataType_FLOAT16:
+      return 2;
+#if (!defined(DISABLE_FLOAT8_TYPES) && (CUDA_VERSION >= 11080))
+    case ONNX_NAMESPACE::TensorProto_DataType_FLOAT8E4M3FN:
+    case ONNX_NAMESPACE::TensorProto_DataType_FLOAT8E5M2:
+      return 1;
+#endif
+    default:
+      ORT_THROW("Unexpected element_type=", element_type, ".");
+  }
+}
+
+void GemmFloat8::set(const TensorShape& a_shape, const TensorShape& b_shape,
+                     int& M, int& N, int& K, int& lda, int& ldb, int& ldd, bool row_major) const {
+  constexpr int ir = 0;
+  constexpr int ic = 1 - ir;
+  if (transA_ && !transB_) {  // TN
```

Comment:
nit: refine the code to make it cleaner, for example: 
int m_idx = transA_? 1: 0; 
int k_idx = 1 - m_idx; 
int n_idx = transB_? 0 : 1;
M = static_cast<int>(a_shape[m_idx]);
K = static_cast<int>(a_shape[k_idx]);
N = static_cast<int>(b_shape[n_idx]);
lda = static_cast<int>(a_shape[row_major ? ic : ir]);
ldb = static_cast<int>(b_shape[row_major ? ic : ir]);
ldd = static_cast<int>(row_major? N : M);

As mentioned, row_major is unnessary.